umbral/pre.py

```diff
@@ -363,8 +368,15 @@ def split_rekey(delegating_privkey: UmbralPrivateKey, signer: Signer,
 
         u1 = rk * u
 
-        kfrag_validity_message = bytes().join(
-            bytes(material) for material in (id, pubkey_a_point, pubkey_b_point, u1, ni, xcoord))
+        pubkey_size = UmbralPublicKey.expected_bytes_length(curve=params.curve)
+        if not sign_delegating_key:
+            delegating_pubkey = b'\x00' * pubkey_size
+        if not sign_receiving_key:
+            receiving_pubkey = b'\x00' * pubkey_size
+
+        validity_input = (id, delegating_pubkey, receiving_pubkey, u1, ni, xcoord)
```

Comment:
Perhaps nit-picking here but let's be careful with name shadowing the builtin functions (`id` in this case), It can lead to some unnecessary code fragility.